File: binarysearch.c
#include<stdio.h>
#include<math.h>

int binary_search(int low,int high, int arr[],int search);

int main(){

    int low = 0;
    int high = 9;
    int arr[10] = {0,1,2,3,4,5,6,7,8,9};
    int search = 0;
    printf("%d",binary_search(low,high,arr,search));

    return 0;
}

int binary_search(int low, int high, int arr[],int search){

    int mid = 0;
    //repeat until low<=high

    while (low<=high)
    {
        //compute mid value
        mid = (high + low) / 2;
        //check if middle index is search value
        if (search == arr[mid]){
            return mid;
        }

        //check for other condition
        if (search > arr[mid]){
            low = mid+1;
        }
        if(search < arr[mid]){
            high = mid -1;
        }
        
    }
    
    //if value is not found in array return -1
    return -1;
}

File: binarysearch.py
def binary_Search(low,high,arr,data):
    #go until condition true low<=high
    while low<=high:
        #calculate mid value
        mid = high + low //2
        #check if mid value is your data
        if data == arr[mid]:
            return mid
        #chech whether data is greater than mid the add 1 to mid and made it low
        if data > arr[mid]:
            low = mid +1
        #else make hig mid -1
        if data < arr[mid]:
            high = mid -1
    #if value is not found return -1
    return-1

arr = [5,8,6,9,2,5]
arr.sort()
low = 0
high = len(arr)-1
data = 6
print(high)
print(binary_Search(low,high,arr,data))